src/api/mc_worker_api.erl

```diff
@@ -177,44 +299,134 @@ count(Connection, Query) ->
   {true, #{<<"n">> := N}} = command(Connection, Query),
   trunc(N). % Server returns count as float
 
-%% @doc Create index on collection according to given spec.
+%% @doc Create index on collection according to given spec. This function does
+%% not work if you have configured the driver to use the new version of the
+%% protocol with application:set_env(mongodb, use_legacy_protocol, false). In
+%% that case you can call the createIndexes
+%% (https://www.mongodb.com/docs/manual/reference/command/createIndexes/#mongodb-dbcommand-dbcmd.createIndexes)
+%% command using the `mc_worker_api:command/2` function instead. 
+%%
 %%      The key specification is a bson documents with the following fields:
 %%      IndexSpec      :: bson document, for e.g. {field, 1, other, -1, location, 2d}, <strong>required</strong>
 -spec ensure_index(pid(), colldb(), bson:document()) -> ok | {error, any()}.
 ensure_index(Connection, Coll, IndexSpec) ->
-  mc_connection_man:request_worker(Connection, #ensure_index{collection = Coll, index_spec = IndexSpec}).
+    case mc_utils:use_legacy_protocol(Connection) of
+        true ->
+            mc_connection_man:request_worker(Connection,
+                                             #ensure_index{collection = Coll,
+                                                           index_spec = IndexSpec});
+        false -> 
+           erlang:error({error, <<"This function does not work when one have specified application:set_env(mongodb, use_legacy_protocol, false). Call the createIndexes command using mc_worker_api:command/2 instead.">>}) 
```

Comment:
Q: would it make sense to return an `{error, _}` instead of calling `error(_)`?

Reply:
I think it makes sense as it makes it more consistent with the rest of the API. Fixed in commit below. One problem however is that there is a danger that old code that upgrade this driver might miss this change more easily if we don't raise an error. Maybe I should change it back to raising an error or is it enough to clearly document this change in the README? What do you think?

Reply:
That's a good point.  

On the one hand, assuming that client code calling this function is sufficiently tested, it is more convenient to the callers to have this fail loudly.  

On the other hand, client code will also have to put `try/catch`es everywhere, and I think it's better to return errors as values.  Maybe we could indicate this change in the readme and/or in the changelog and hope users will read it :joy_cat:

Reply:
Searching for `ensure_index` in our CE and EE (4.4) repos show no results, so from our side it seems safe :smiley_cat:

Reply:
I have updated the README with some info about the change in the ensure_index functions.